### DISCUSSION TITLE
PS2, Part (h)

### DISCUSSION THREAD

**Discussion by lucashkaplan** [spamegg1/Math-for-CS-solutions#2] [Q&A] [answered]
Hey all, I had a question regarding PS2, part (h). Everything in the solution makes sense to me, except for one part. I don't understand why if a language S is boring, then so is NOT(S). Any explanation would be greatly appreciated.

**Comment by spamegg1** [comment, marked as answer]
**Definition:** $S$ is boring when either $S$ or $\overline{S}$ is 0-finite.

1. Assume $S$ is boring. 
[We want to show $\overline{S}$ is boring. In other words we want to show either $\overline{S}$ or $\overline{\overline{S}}$ is 0-finite.]
2. By 1 and the definition, either $S$ or $\overline{S}$ is 0-finite.
3. By 2, either $\overline{\overline{S}}$ or $\overline{S}$ is 0-finite. [Because $S = \overline{\overline{S}}$.]
4. Rewriting 3, either $\overline{S}$ or $\overline{\overline{S}}$ is 0-finite.
5. By 4 and the definition, $\overline{S}$ is boring.
